src/math.rs

```diff
@@ -185,25 +185,22 @@ pub fn crt(r: &[i64], m: &[i64]) -> (i64, i64) {
 ///
 /// assert_eq!(math::floor_sum(6, 5, 4, 3), 13);
 /// ```
+#[allow(clippy::many_single_char_names)]
 pub fn floor_sum(n: i64, m: i64, mut a: i64, mut b: i64) -> i64 {
+    assert!(0 <= n && n < 1i64 << 32);
```

Comment:
Could you just replace it with something like `matches!(n, 0..1 << 32)`? This is slightly better than `(0..1 << 32).contains(&n)` because it does not involve taking a reference to `n`, which may avoid constant-factor performance degradation.

Reply:
Since the implementation of `matches!` macro seems to be insufficient, it causes error on parsing. Should I replace it with `(0..1<<32).contains(&n)` ?

```
% cargo build
   Compiling ac-library-rs v0.1.0 (/home/user/xxx/ac-library-rs)
error: no rules expected the token `<<`
   --> src/math.rs:190:33
    |
190 |     assert!(matches!(n, 0..1i64 << 32));
    |                                 ^^ no rules expected this token in macro call

error: no rules expected the token `<<`
   --> src/math.rs:191:33
    |
191 |     assert!(matches!(m, 1..1i64 << 32));
    |                                 ^^ no rules expected this token in macro call

error: aborting due to 2 previous errors

error: could not compile `ac-library-rs`.

To learn more, run the command again with --verbose.
```